### PR TITLE
Datasource for Cloud Databases Backups and Backup

### PR DESCRIPTION
**Pull request by obai-1**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TEST=./ibm/service/database TESTARGS='-run=TestAccIBMDatabaseBackupsDataSourceBasic'
 ...
=== RUN   TestAccIBMDatabaseBackupsDataSourceBasic
--- PASS: TestAccIBMDatabaseBackupsDataSourceBasic (24.06s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/database	25.198s

```

### CODE REVIEW

File: ibm/service/database/data_source_ibm_database_backups_test.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+func TestAccIBMDatabaseBackupsDataSourceBasic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { acc.TestAccPreCheck(t) },
+		Providers: acc.TestAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIBMDatabaseBackupsDataSourceConfigBasic(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.ibm_database_backups.database_backups", "id"),
+					resource.TestCheckResourceAttrSet("data.ibm_database_backups.database_backups", "id"),
```
Comment (omaraibrahim) [inline ibm/service/database/data_source_ibm_database_backups_test.go:24]: This test shouldn't be duplicated.

File: ibm/service/database/data_source_ibm_database_backups_test.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+func TestAccIBMDatabaseBackupsDataSourceBasic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { acc.TestAccPreCheck(t) },
+		Providers: acc.TestAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIBMDatabaseBackupsDataSourceConfigBasic(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.ibm_database_backups.database_backups", "id"),
```
Comment (omaraibrahim) [inline ibm/service/database/data_source_ibm_database_backups_test.go:23]: `id` should be `deployment_id` based on the code above.

File: ibm/service/database/data_source_ibm_database_backups_test.go (hunk 1 above)
Comment (omaraibrahim) [inline ibm/service/database/data_source_ibm_database_backups_test.go:24]: Other things in the schema aren't being tested.

File: ibm/service/database/data_source_ibm_database_backup_test.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+func TestAccIBMDatabaseBackupDataSourceBasic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { acc.TestAccPreCheck(t) },
+		Providers: acc.TestAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIBMDatabaseBackupDataSourceConfigBasic(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.ibm_database_backup.database_backup", "id"),
```
Comment (omaraibrahim) [inline ibm/service/database/data_source_ibm_database_backup_test.go:23]: Should id be tested?

File: ibm/service/database/data_source_ibm_database_backup_test.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+					resource.TestCheckResourceAttrSet("data.ibm_database_backup.database_backup", "backup_id"),
```
Comment (omaraibrahim) [inline ibm/service/database/data_source_ibm_database_backup_test.go:24]: Other parts of the schema aren't being tested.

File: ibm/service/database/data_source_ibm_database_backup_test.go (hunk 3 above)
Comment (obai-1) [reply ibm/service/database/data_source_ibm_database_backup_test.go]: No, removed

**Comment by obai-1** [comment]
@omaraibrahim addressed those comments

**Review by omaraibrahim — COMMENTED** [review]
LGTM

**Pull request merged**